lib/cli_executor.rb

```diff
@@ -305,6 +309,9 @@ def exec(user, key, opts={})
 
       add_proxy_env_opt(user.env, opts)
       cli_tool = tool_from_opts!(opts)
+      if CENSOR_LOGS.any? { |cmd| key.match?(cmd) }
+        opts << [:_quiet, true]
+      end
       executor(cli_tool: cli_tool).
         run(key, Common::Rules.merge_opts(logged_users[user.id], opts))
     end
```

Comment:
After more looking at the stacks from https://github.com/openshift/verification-tests/pull/2524/files, checked `executor.run` of lib/rules_command_executor.rb, it has `cmd, cmd_options, exec_options`, it is better place to censor the cmd, cmd_options than here because it can use RE to match the cmd like like get-token, whoami -t, get secret -o yaml/json, etc.

Reply:
But I then searched :response in `lib/local_process.rb`, find that is **even further better** place to censor, we can censor command before `logger.plain(result[:response]...`. Via this way, the shell command can be still printed, which is useful, only shell command result is hidden. Like:
```
          if cmd matched censored commands like get-token, whoami -t, get secret -o yaml/json, etc.
            is_censored_cmd = true
          end
          unless opts[:quiet] || is_censored_cmd
            logger.plain(result[:response], false)
```
Like demo https://github.com/xingxingxia/verification-tests/pull/1/files , its test log: https://...ocp-c1.prod.psi.redhat.com/job/runner-v3-smoke/3760/console , it can censor any auto commands, not only oc.
Could you help to elaborate based on this? (No further time on it; need allocate time on epics test : ) )
Thanks!

Reply:
We can iterate to better and better.
Currently the tests on Prow are all disabled/removed due to those sensitive info in the logs.